File: src/main/java/com/dyrosoft/kvbparser/parser/AbstractRxHtmlParserFunc.java
package com.dyrosoft.kvbparser.parser;

import rx.Single;
import rx.functions.Func1;

abstract class AbstractRxHtmlParserFunc<T> implements Func1<String, Single<T>> {

    @Override
    public Single<T> call(final String html) {
        return Single.just(parse(html));
    }

    protected abstract T parse(final String html);
}

File: src/main/java/com/dyrosoft/kvbparser/parser/AbstractRxHtmlParserFunc.kt
package com.dyrosoft.kvbparser.parser

import rx.Single
import rx.functions.Func1

abstract class AbstractRxHtmlParserFunc<T> : Func1<String, Single<T>> {

    override fun call(html: String): Single<T> = Single.just(parse(html))

    protected abstract fun parse(html: String): T
}
